### PR TITLE
Don't mutate container disk permissions

### PR DESCRIPTION
**Pull request by davidvossel**
If we mutate the permissions of the disk, then we cause the disk to be copied due to the container's layered filesystem. A container disk should never be altered. 

The disk either has the correct permissions necessary for qemu to access it, or it doesn't. If it doesn't, then the vmi will fail to boot. There's no action we can take to change or modify that on our end. 

```release-note
NONE
```

### CODE REVIEW

**Review by rmohr — CHANGES_REQUESTED** [review]
We know how the container disk in tests looks like. Can we extend the test to vheck that? Let's see that we get a failing test if the permission amor ownership changes.

I think this should be backported to every release with this bug. 

/kind bug

File: tests/container_disk_test.go [hunk 1]
```diff
@@ -250,17 +250,6 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 
 				By("Checking the writable Image Octal mode")
 				Expect(strings.Trim(writableImageOctalMode, "\n")).To(Equal("640"), "Octal Mode of writable Image should be 640")
-
-				readonlyImageOctalMode, err := tests.ExecuteCommandOnPod(
```
Comment (danielBelenky) [inline tests/container_disk_test.go:254]: Maybe add a test that specifically ensures that we don't try to apply any modifications to the container-disk in such cases?

**Review by danielBelenky — CHANGES_REQUESTED** [review]

**Comment by rmohr** [comment]
/cc @xpivarc @jean-edouard 

FYI, related to #3819.

**Review by jean-edouard — APPROVED** [review]

**Comment by davidvossel** [comment]
/hold

After further investigation, I'm not sure this is really an issue

@rmohr After investigating this further, i'm not actually seeing the chmod cause any additional usage of storage space. I created a container disk with a 4gb disk and ran it in a replicaset of size 5. There was no increase in storage usage beyond the initial pull of the 4gb image.

maybe inode changes don't actually result in a modification to the file itself. The "modify" timestamp never changes on the disk image even after the chmod.

**Comment by rmohr** [comment]
> maybe inode changes don't actually result in a modification to the file itself. The "modify" timestamp never changes on the disk image even after the chmod.

I am not sure we can make this asumption in general. Maybe some optimizations exist, but if I read for instance the `overlayfs` documentation (as said can be different in newer implementations): https://www.kernel.org/doc/Documentation/filesystems/overlayfs.txt, it says:

```
The copy_up process first makes sure that the containing directory
exists in the upper filesystem - creating it and any parents as
necessary.  It then creates the object with the same metadata (owner,
mode, mtime, symlink-target etc.) and then if the object is a file, the
data is copied from the lower to the upper filesystem.  Finally any
extended attributes are copied up.
```

which highlights that `mode` is part of what is tracked as content of the object.

**Comment by xpivarc** [comment]
```
Metadata only copy up
--------------------

When metadata only copy up feature is enabled, overlayfs will only copy
up metadata (as opposed to whole file), when a metadata specific operation
like chown/chmod is performed. Full file will be copied up later when
file is opened for WRITE operation.
```

This is probably why the file is not being copied. But we should probably also consider other options than `overlayfs`.

**Comment by rmohr** [comment]
> This is probably why the file is not being copied. But we should probably also consider other options than `overlayfs`.

Yes, thanks for digging that paragraph out. I think we should be on the save side here, since even in `overlayfs` it seems to be a features which is not necessarily enabled.

**Comment by davidvossel** [comment]
@rmohr i changed things around a bit with regards to how our container disk images are built and tested.

I now set 444 at built time for our container disks. This should ensure we'll encounter issues in our functional tests if our runtime ever tries to mutate the disks. I also restored the vcheck in our functional test to verify that the mode is as we expect it based on what our build system is doing.

This verification doesn't guarantee us that we don't mutate the container disk image, but it at least verifies that the file permissions are what we'd expect.

**Pull request merged**